src/diffusers/models/attention.py

```diff
@@ -84,6 +84,420 @@ def forward(self, x: torch.Tensor, objs: torch.Tensor) -> torch.Tensor:
+
+        # 2. Cross-Attn
+        self.norm3 = FP32_Layernorm(dim, norm_eps, norm_elementwise_affine)
+
+        self.attn2 = HunyuanDiTCrossAttention(dim, text_dim, num_heads=num_attention_heads, qkv_bias=True, qk_norm=qk_norm)
+
+        # 3. Feed-forward
+        self.norm2 = FP32_Layernorm(dim, norm_eps, norm_elementwise_affine)
+
+        ### NOTE: do not switch norm2 and norm3, otherwise will load wrong key when using pretrained model!
+
+        #print('mlp hidden dim:', ff_inner_dim)
+        self.ff = FeedForward(
+            dim,
+            dropout=dropout, ### 0.0
+            activation_fn=activation_fn, ### approx GeLU
+            final_dropout=final_dropout, ### 0.0
+            inner_dim=ff_inner_dim, ### int(dim * mlp_ratio)
+            bias=ff_bias,
+        )
+
+        # 4. Skip Connection
+        if skip:
+            self.skip_norm = FP32_Layernorm(2 * dim, norm_eps, elementwise_affine=True)
+            self.skip_linear = nn.Linear(2 * dim, dim)
+        else:
+            self.skip_linear = None
+
+        # 5. SDXL-style modulation with add
+        self.default_modulation = nn.Sequential(
+            FP32_SiLU(),
+            nn.Linear(dim, dim, bias=True)
+        )
+
+        # let chunk size default to None
+        self._chunk_size = None
+        self._chunk_dim = 0
+
+    def set_chunk_feed_forward(self, chunk_size: Optional[int], dim: int = 0):
+        # Sets chunk feed-forward
+        self._chunk_size = chunk_size
+        self._chunk_dim = dim
+
+    def forward(
+        self,
+        hidden_states: torch.Tensor,
+        encoder_hidden_states: Optional[torch.Tensor] = None,
+        timestep: Optional[torch.LongTensor] = None,
+        freq_cis_img = None,
+        skip=None
+    ) -> torch.Tensor:
+
+        # Notice that normalization is always applied before the real computation in the following blocks.
+        # 0. Long Skip Connection
+        if self.skip_linear is not None:
+            cat = torch.cat([hidden_states, skip], dim=-1)
+            cat = self.skip_norm(cat)
+            hidden_states = self.skip_linear(cat)
+        
+        #print('x:', hidden_states[0])
+        # 1. Self-Attention
+        norm_hidden_states = self.norm1(hidden_states) ### checked: self.norm1 is correct
+        shift_msa = self.default_modulation(timestep).unsqueeze(dim=1)
+        attn_inputs = (norm_hidden_states + shift_msa, freq_cis_img,)
+        attn_output = self.attn1(*attn_inputs)[0]
+        hidden_states = hidden_states + attn_output
+        #print('x:', hidden_states[0])
+
+        # 2. Cross-Attention
+        cross_inputs = (
+            self.norm3(hidden_states), encoder_hidden_states, freq_cis_img
+        )
+        hidden_states = hidden_states + self.attn2(*cross_inputs)[0]
+        #print('x:', hidden_states[0])
+
+        # FFN Layer ### NOTE: do not switch norm2 and norm3, otherwise will load wrong key when using pretrained model!
+        mlp_inputs = self.norm2(hidden_states)
```

Comment:
This reads a little haphazard to me. Could you modify the state dictionary so that it's not the case?

Reply:
ok